### PR TITLE
Enable Qlora scripts on Intel GPUs

### PR DESCRIPTION
**Pull request by abhilash1910**
Motivation:
- Thanks for the qlora repo, this PR is for adding intel gpu support for quantized inference in the qlora script.
- There is an internal progress on B&B enabling for intel gpus which will make the support complete.

@artidoro  could you please review . Would appreciate your thoughts on this. Thanks

### CODE REVIEW

File: qlora.py [hunk 1]
```diff
@@ -303,6 +334,10 @@ def get_accelerate_model(args, checkpoint_dir):
             print('='*80)
             print('Your GPU supports bfloat16, you can accelerate training with the argument --bf16')
             print('='*80)
+            
+    if compute_dtype == torch.float16 and (is_ipex_available() and torch.xpu.is_available()):
+        compute_dtype = torch.bfloat16
+        print('Intel XPU does not supports float16 yet, you can accelerate training with the argument --bf16')
```
Comment (artidoro) [inline qlora.py:340]: Thanks for the PR!
Does this mean that you are effectively switching to bf16 but printing a message that says the user should be switching?
If so, maybe change the message saying "Intel XPU does not support float16 yet, so switching to bfloat16".

**Review by artidoro — COMMENTED** [review]
Thanks for the PR!

File: qlora.py (hunk 1 above)
Comment (abhilash1910) [reply qlora.py]: Thanks for the pointer , modified it.

**Comment by artidoro** [comment]
So it looks good to me, however, I cannot test the code as I don't have an Intel GPU. Could you let me know how you verified that it works?
@TimDettmers  do you have thoughts on adding this to the repo? It could be helpful to people using Intel GPUs and it seems relatively few changes.

**Comment by abhilash1910** [comment]
@artidoro  @TimDettmers at Intel we are planning to integrate our kernel code for the quantization cu kernels . This is a work in progress internally and after validation I will flag a PR at b&b. The pythonic frontend (here) has relatively easier changes but the core of the task is in the kernel adaptation for our gpu devices  which we are integrating.

**Comment by artidoro** [comment]
Sounds good! I am happy to merge this PR. Might be useful to add some documentation too.

**Pull request merged**